Build failure: linux-kernel-latest-htmldocs-6.8.drv

### Steps To Reproduce

Steps to reproduce the behavior:
1. build linuxPackages_latest

### Build log

```
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/qq141igg1nrl85g5ssnn86vwwnfcbi62-linux-kernel-latest-htmldocs-6.8.drv^*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/snqmwdskgnbbg45wc4492dp7r64xrgak-linux-6.8.tar.xz
source root is linux-6.8
setting SOURCE_DATE_EPOCH to timestamp 1710103089 of file linux-6.8/virt/lib/irqbypass.c
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
patching script interpreter paths in Documentation/sphinx/parse-headers.pl scripts/get_abi.pl scripts/get_feat.pl scripts/kernel-doc scripts/sphinx-pre-install
Documentation/sphinx/parse-headers.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/2l02m7b3p0f8fk45qiipmdpzc2myl09q-perl-5.38.2/bin/perl"
scripts/get_abi.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/2l02m7b3p0f8fk45qiipmdpzc2myl09q-perl-5.38.2/bin/perl"
scripts/get_feat.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/2l02m7b3p0f8fk45qiipmdpzc2myl09q-perl-5.38.2/bin/perl"
scripts/kernel-doc: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/2l02m7b3p0f8fk45qiipmdpzc2myl09q-perl-5.38.2/bin/perl"
scripts/sphinx-pre-install: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/2l02m7b3p0f8fk45qiipmdpzc2myl09q-perl-5.38.2/bin/perl"
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
build flags: SHELL=/nix/store/4vzal97iq3dmrgycj8r0gflrh51p8w1s-bash-5.2p26/bin/bash htmldocs
/nix/store/4vzal97iq3dmrgycj8r0gflrh51p8w1s-bash-5.2p26/bin/bash: line 1: ./tools/net/ynl/ynl-gen-rst.py: cannot execute: required file not found
make[2]: *** [Documentation/Makefile:112: Documentation/networking/netlink_spec/devlink.rst] Error 127
make[1]: *** [/build/linux-6.8/Makefile:1705: htmldocs] Error 2
make: *** [Makefile:240: __sub-make] Error 2
```

### Additional context

I tried to avoid it with
- run garbage collection
- set `boot.kernelPackages = pkgs.linuxKernel.packages.linux_6_7;` instead of `boot.kernelPackages = pkgs.linuxPackages_latest;`
- run `sudo nix-store --verify --check-contents --repair`

but these did not help.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.9, NixOS, 24.05 (Uakari), 24.05.20240309.3030f18`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/nra828scc8qs92b9pxra5csqzffb6hpl-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/kernel/htmldocs.nix
+++ b/pkgs/os-specific/linux/kernel/htmldocs.nix
@@ -18,7 +18,8 @@ stdenv.mkDerivation {
   postPatch = ''
     patchShebangs \
       Documentation/sphinx/parse-headers.pl \
-      scripts/{get_abi.pl,get_feat.pl,kernel-doc,sphinx-pre-install}
+      scripts/{get_abi.pl,get_feat.pl,kernel-doc,sphinx-pre-install} \
+      tools/net/ynl/ynl-gen-rst.py
   '';
 
   FONTCONFIG_FILE = makeFontsConf {
@@ -31,6 +32,7 @@ stdenv.mkDerivation {
     perl
     python3.pkgs.sphinx
     python3.pkgs.sphinx-rtd-theme
+    python3.pkgs.pyyaml
     which
   ];
 

